toolchain/check/testdata/impl/incomplete.carbon

```diff
@@ -0,0 +1,862 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/none.carbon
+// TODO: Add ranges and switch to "--dump-sem-ir-ranges=only".
+// EXTRA-ARGS: --dump-sem-ir-ranges=if-present
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/impl/incomplete.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/impl/incomplete.carbon
+
+// --- fail_empty_struct.carbon
+library "[[@TEST_NAME]]";
+
+interface I;
+
+// Requires I identified.
+impl {} as I;
+
```

Comment:
```suggestion
// --- forward_decl.carbon
library "[[@TEST_NAME]]";

interface I;

// Requires I identified.
impl {} as I;

// --- fail_empty_struct.carbon
library "[[@TEST_NAME]]";

interface I;

impl {} as I;

```
... so that we have a non `fail_` test for forward-declaring an impl of a forward-declared interface.

Reply:
If we split it, the first test will fail with "impl declaration was never defined", so it still ends up being a fail_ test.